src/helpers.jl

```diff
@@ -0,0 +1,407 @@
+        return true
+    end
+    false
+end
+ismeson(p::Union{Particle, Integer}) = ismeson(convert(PDGID, p))
+
+
+function isbaryon(p::PDGID)
+    abspdgid = abs(p.value)
+
+    abspdgid <= 100 && return false
+
+    # Special case of proton and neutron:
+    # needs to be checked first since isstandard() is false for nuclei
+    abspdgid ∈ [1000000010, 1000010010] && return true
+
+    !isstandard(p) && return false
+    0 < fundamentalid(p) <= 100 && return false
+
+    # Old codes for diffractive p and n (MC usage)
+    abspdgid ∈ [2110, 2210] && return true
+
+    p.Nj > 0 && p.Nq3 > 0 && p.Nq2 > 0 && p.Nq1 > 0 && return true
+
+    (isRhadron(p) || ispentaquark(p)) && return false
+
+    false
+end
+isbaryon(p::Union{Particle, Integer}) = isbaryon(convert(PDGID, p))
+
+function ishadron(p::PDGID)
+    abs(p.value) ∈ [1000000010, 1000010010] && return true
+    !isstandard(p) && return false
+    ismeson(p) && return true
+    isbaryon(p) && return true
+    ispentaquark(p) && return true
+    isRhadron(p) && return true
+    false
+end
+ishadron(p::Union{Particle, Integer}) = ishadron(convert(PDGID, p))
+
+"""
+$(SIGNATURES)
+
+An R-hadron is of the form 10abcdj, 100abcj, or 1000abj,
+where j = 2J + 1 gives the spin; b, c, and d are quarks or gluons;
+and a (the digit following the zero's) is a SUSY particle.
+"""
+function isRhadron(p::PDGID)
+    !isstandard(p) && return false
+    (p.N != 1 || p.Nr != 0) && return false
+    isSUSY(p) && return false
+    # All R-hadrons have at least 3 core digits
+    (p.Nq2 == 0 || p.Nq3 == 0 || p.Nj == 0) && return false
+    true
+end
+isRhadron(p::Union{Particle, Integer}) = isRhadron(convert(PDGID, p))
+
+
+"""
+$(SIGNATURES)
+
+Fundamental SUSY particles have N = 1 or 2.
+"""
+function isSUSY(p::PDGID)
+    !isstandard(p) && return false
+    p.N != 1 && p.N != 2 && return false
+    p.Nr != 0 && return false
+    fundamentalid(p) == 0 && return false
+    true
+end
+isSUSY(p::Union{Particle, Integer}) = isSUSY(convert(PDGID, p))
+
+
+"""
+$(SIGNATURES)
+
+Pentaquark IDs are of the form +/- 9 Nr Nl Nq1 Nq2 Nq3 Nj,
+where Nj = 2J + 1 gives the spin and Nr Nl Nq1 Nq2 Nq3 denote the quark numbers
+in order Nr >= Nl >= Nq1 >= Nq2 and Nq3 gives the antiquark number.
+"""
+function ispentaquark(p::PDGID)
+    !isstandard(p) && return false
+    p.N != 9 && return false
+    (p.N == 9 || p.Nr == 0) && return false
+    (p.Nj == 9 || p.Nl == 0) && return false
+    p.Nq1 == 0 && return false
+    p.Nq2 == 0 && return false
+    p.Nq3 == 0 && return false
+    p.Nj == 0 && return false
+    p.Nq2 > p.Nq1 && return false
+    p.Nq1 > p.Nl && return false
+    p.Nl > p.Nr && return false
+    true
+end
+ispentaquark(p::Union{Particle, Integer}) = ispentaquark(convert(PDGID, p))
+
+isgaugebosonorhiggs(p::PDGID) = 21 <= abs(p.value) <= 40
+isgaugebosonorhiggs(p::Union{Particle, Integer}) = isgaugebosonorhiggs(convert(PDGID, p))
+issmgaugebosonorhiggs(p::PDGID) = abs(p.value) == 24 || 21 <= p.value <= 25
```

Comment:
Is there a special reason to have an individual check for 24. I guess it should be contained in the range [21,25].

Reply:
It's `abs`, so it also checks for -24 😉

Reply:
Fail ... sorry overlooked that `abs` :see_no_evil:

Reply:
No worries, I was confused as hell many times. 🙈